File: java/src/main/java/com/authdog/types/Session.java
package com.authdog.types;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Session information.
 */
public final class Session {
    /**
     * Remaining seconds in session.
     */
    @JsonProperty("remainingSeconds")
    private int remainingSeconds;

    /**
     * Default constructor.
     */
    public Session() {
    }

    /**
     * Constructor with parameters.
     * @param remainingSecondsParam Remaining seconds in session
     */
    public Session(final int remainingSecondsParam) {
        this.remainingSeconds = remainingSecondsParam;
    }

    /**
     * Get remaining seconds in session.
     * @return Remaining seconds in session
     */
    public int getRemainingSeconds() {
        return remainingSeconds;
    }

    /**
     * Set remaining seconds in session.
     * @param remainingSecondsParam Remaining seconds in session
     */
    public void setRemainingSeconds(final int remainingSecondsParam) {
        this.remainingSeconds = remainingSecondsParam;
    }
}

File: scala/src/main/scala/com/authdog/types/Session.scala
package com.authdog.types

import io.circe.{Decoder, Encoder, HCursor, Json}
import io.circe.syntax._

/**
 * Session information
 */
case class Session(
  remainingSeconds: Int
)

object Session {
  implicit val decoder: Decoder[Session] = (c: HCursor) => for {
    remainingSeconds <- c.downField("remainingSeconds").as[Int]
  } yield Session(remainingSeconds)
  
  implicit val encoder: Encoder[Session] = (session: Session) => Json.obj(
    "remainingSeconds" -> session.remainingSeconds.asJson
  )
}
